packages/teams-js/src/public/authentication.ts

```diff
@@ -159,46 +127,25 @@ export namespace authentication {
   }
 
   function authenticateHelper(apiVersionTag: string, authenticateParameters: AuthenticateParameters): Promise<string> {
-    return new Promise<string>((resolve, reject) => {
-      if (
-        GlobalVars.hostClientType === HostClientType.desktop ||
-        GlobalVars.hostClientType === HostClientType.android ||
-        GlobalVars.hostClientType === HostClientType.ios ||
-        GlobalVars.hostClientType === HostClientType.ipados ||
-        GlobalVars.hostClientType === HostClientType.macos ||
-        GlobalVars.hostClientType === HostClientType.rigel ||
-        GlobalVars.hostClientType === HostClientType.teamsRoomsWindows ||
-        GlobalVars.hostClientType === HostClientType.teamsRoomsAndroid ||
-        GlobalVars.hostClientType === HostClientType.teamsPhones ||
-        GlobalVars.hostClientType === HostClientType.teamsDisplays ||
-        GlobalVars.hostClientType === HostClientType.surfaceHub
-      ) {
-        // Convert any relative URLs into absolute URLs before sending them over to the parent window.
-        const link = document.createElement('a');
-        link.href = authenticateParameters.url;
-        // Ask the parent window to open an authentication window with the parameters provided by the caller.
-        resolve(
-          sendMessageToParentAsync<[boolean, string]>(apiVersionTag, 'authentication.authenticate', [
-            link.href,
-            authenticateParameters.width,
-            authenticateParameters.height,
-            authenticateParameters.isExternal,
-          ]).then(([success, response]: [boolean, string]) => {
-            if (success) {
-              return response;
-            } else {
-              throw new Error(response);
-            }
-          }),
-        );
-      } else {
-        // Open an authentication window with the parameters provided by the caller.
-        authHandlers = {
-          success: resolve,
-          fail: reject,
-        };
-        openAuthenticationWindow(authenticateParameters);
-      }
+    return new Promise<string>((resolve) => {
```

Comment:
Now that we know this code is needed for some specific browsers, it would be good to change this PR to instead put in a thorough comment documenting why the `window.open` flow is needed and how it works.